src/bloom.cpp

```diff
@@ -34,12 +34,12 @@ CBloomFilter::CBloomFilter(unsigned int nElements, double nFPRate, unsigned int
      * Again, we ignore filter parameters which will create a bloom filter with more hash functions than the protocol limits
      * See https://en.wikipedia.org/wiki/Bloom_filter for an explanation of these formulas
      */
-    isFull(false),
     isEmpty(false),
     nHashFuncs(min((unsigned int)(vData.size() * 8 / nElements * LN2), MAX_HASH_FUNCS)),
     nTweak(nTweakIn),
     nFlags(nFlagsIn)
 {
+    isFull = vData.size() == 0;
```

Comment:
How about just adding a call to UpdateEmptyFull at the end of every explicit constructor, and at the end of SerializationOp when ser_action is to deserialize?  And possibly removing resulting redundant calls.

clear() still needs @awemany's fix. As the author of that function, I don't remember considering that the initial fullness state is true for an empty filter.